### ISSUE TITLE
[Feature request] Showing raw addresses in Correspondents column

### ISSUE THREAD

**Issue by kanlukasz** [lkosson/full-address-column#2]
Hello,
Could you please add an option (column) to show raw addresses in the Correspondents column?

**Comment by lkosson** [comment]
I'm not quite sure how to determine whether message is inbound or outbound from TB API, as original Correspondents column does - especially for folders containing both outgoing and incoming messages, for instance gmail's "All Mail" folder.

**Comment by kanlukasz** [comment]
Oh, ok, I didn't know it's complicated.
Unfortunately, I have no idea how to help because I mainly work with PHP 😞

**Comment by AdamTT** [comment]
> 
> 
> I'm not quite sure how to determine whether message is inbound or outbound from TB API, as original Correspondents column does - especially for folders containing both outgoing and incoming messages, for instance gmail's "All Mail" folder.

Hi,
I'm not java (or else) programmer, so I'm not sure if U can code it, but my idea is:
1. read address twice
   - first from 'Corespondents' column,
   - second from 'recipient' column for example.
2. compare those 2 addresses
   - if they are the same - it is incoming message,
   - else it is outcoming message.
3. Display proper icon followed  by address from  'Corespondents' column,

If icon is to complicated then maybe it is enough to display:
 '> ' or '>>  ' for out- and 
 '< ' or '<<  ' for incoming nessage

Anyway, thank You for this 'must have' ext.

**Comment by lkosson** [comment]
The problem is I don't know how or if it is even possible to fetch content of "Corespondents" column programmatically.

**Comment by AdamTT** [comment]
Well, I understand. 
I guess U read addresses direct from particular message headers, right?

**Comment by lkosson** [comment]
More or less - yes. The addon uses `nsIMsgDBHdr` as defined in https://hg.mozilla.org/comm-central/file/tip/mailnews/base/public/nsIMsgHdr.idl

**Comment by AdamTT** [comment]
I see nsIMsgDBHdr reads only few headers. It is not able to check (for example) presence of header 'Delivered-To:' or 'Received:'?

**Comment by lkosson** [comment]
I don't think so. There are `getProperty` and `getStringProperty` methods, but they don't seem to have access to arbitrary MIME header - most likely because they operate on Thunderbird's internal message database, not a raw e-mail content.

**Comment by AdamTT** [comment]
OK, I give up. 
Just promise me you will leave the note here if u discover a solution in the future.
PS. Mówisz po polsku, nie? :)

**Comment by lkosson** [comment]
Sure will. I'll leave this issue open as long as it is relevant.
PS. Tak :-)
